### ISSUE TITLE
Add support for example_spec, similar to test_spec but for examples

### ISSUE THREAD

**Issue by jverkoey** [CocoaPods/CocoaPods#8121]
🌈

* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

### Context

The [Material Components for iOS](https://github.com/material-components/material-components-ios) project makes heavy use of CocoaPods subspecs to deploy individual user interface components. Each of our components consists of source, tests, and examples.

We use CocoaPods exclusively to manage our Xcode project so that we can minimize Xcode merge conflicts that often arise during team development. We use local CocoaPods podspecs with wildcards to pull in all of the source, examples, and unit test code as a set of development pods.

For example, all of our examples are pulled in via a local [MaterialComponentsExamples](https://github.com/material-components/material-components-ios/blob/develop/MaterialComponentsExamples.podspec) podspec. Similarly, we pull in all of our unit tests using a local [MaterialComponentUnitTests](https://github.com/material-components/material-components-ios/blob/develop/MaterialComponentsUnitTests.podspec) podspec, which we are in the process of replacing with [test_spec](https://github.com/material-components/material-components-ios/pull/5204).

The `test_spec` functionality is great because it allows us to find the test source right next to the component's source. It also improves [other aspects of testing](https://github.com/material-components/material-components-ios/pull/5204) that make our day-to-day workflows much easier. Thank you for this awesome feature!

### The feature request

Support for an `example_spec` type that has the following behaviors:

- Like `test_spec`, `example_spec` source will not be included in the parent spec's code when adding the parent spec as a dependency. Podfiles will need to include an `:examplespecs` parameter to explicitly add examples to a given app.
- Like `test_spec`, `example_spec` source would depend on the parent spec and include all of its transitive dependencies.
- Like `spec`, `example_spec` will create a library or framework that can be included and referenced from a catalog application. A simple catalog application would set an example or list of examples as the root view controller of the app.

**Comment by dostrander** [comment]
@jverkoey Yes #8104 adds the ability to create an app host based on the DSL, with provided sources and dependencies
